assets/js/src/core/modules/asset/editor/toolbar/workflow-log-modal/hooks/use-workflow.ts

```diff
@@ -32,9 +33,20 @@ export type TransitionType = 'transition' | 'global'
 export const useWorkflow = (): UseWorkflowHookReturn => {
   const { openModal, closeModal, isModalOpen, contextWorkflowDetails, setContextWorkflowDetails } = useContext(WorkflowContext)
   const { id, elementType } = useElementContext()
+
   const { asset } = useAssetDraft(id)
+  const { dataObject } = useDataObjectDraft(id)
+
+  const assetHasWorkflowAvailable = asset?.hasWorkflowAvailable ?? false
+  const objectHasWorkflowAvailable = dataObject?.hasWorkflowAvailable ?? false
+
+  const hasWorkflowAvailable =
+      elementType === 'asset'
+        ? assetHasWorkflowAvailable
+        : elementType === 'data-object'
+          ? objectHasWorkflowAvailable
+          : false
```

Comment:
Can't you use the element draft in that case? Because the workflows will also be available for documents. 
And it would remove the need for all these checks. :)

Reply:
ah super cool! - I had not known about that option. have changed that now :)